src/vanilla/store.ts

```diff
@@ -418,14 +418,17 @@ const buildStore = (
   const readAtom = <Value>(atom: Atom<Value>): Value =>
     returnAtomValue(readAtomState(undefined, atom))
 
-  const getDependents = <Value>(
+  const getMountedOrPendingDependents = <Value>(
     pending: Pending,
     atom: Atom<Value>,
     atomState: AtomState<Value>,
   ): Map<AnyAtom, AtomState> => {
     const dependents = new Map<AnyAtom, AtomState>()
     for (const a of atomState.m?.t || []) {
-      dependents.set(a, getAtomState(a))
+      const aState = getAtomState(a)
+      if (aState.m) {
+        dependents.set(a, aState)
```

Comment:
Could you clarify this section? What specific behavior are you aiming to achieve here?